scripts/sanitycheck

```diff
@@ -2883,6 +2887,18 @@ def parse_arguments():
         which will ultimately disable ccache.
         """
     )
+    parser.add_argument(
+        "--west-flash-option", action="append", default=[],
+        help="""Extra options to pass to the "west flash" command when doing
+        running with --device-testing"
+
+        E.g
+         sanitycheck --device-testing --device-serial /dev/ttyACM0 \
+                     --west-flash-option="--board-id=foobar"
+        will translate to
+         west flash -- --board-id=foobar
+        """
```

Comment:
Shouldn't there be two options?

One option for using west, and another for appending options to it.

What if you want to use west, but don't want any additional options?

Reply:
@SebastianBoe That seems reasonable. 
@nashif  @mbolivar  - does that make sense to you guys?

Reply:
I guess it makes sense? I'm not really sure though; `ninja flash` is exactly "west without additional options", by design...

Reply:
clarification: of you want to do it that way, no objections from me, but it seems a bit strange for the above reason.

Reply:
```python
   "--west-flash [ west_flash_options ]"
help="uses west instead of ninja or make to flash"
```

Reply:
I'm not sure how to interpret @marc-hb's comment. Are you saying you want me to rename the existing flag from --west-flash-option -> --west-flash. Remove the old option in favor of the method you are showing here?(which isn't exactly python argparse friendly). or make "--west-flash" a "store_true" option that then explicitly uses west-flash and any options we may have added via --west-flash-option ?

Reply:
I'm suggesting: `"--west-flash", nargs=?, default=None, const=[]`

(`default` can probably be omitted)